### PR TITLE
Add functionality to apply lossless image optimization

### PR DESCRIPTION
**Pull request by tmp-hallenser**
Implement lossless image optimization (#630) 
Uploaded original photo is not touched, only medium, small and thumbnails.

Requirements: jpegoptim, optipng, pngquant, gifsicle, webp

### CODE REVIEW

**Review by ildyria — COMMENTED** [review]
LGTM.

Do we want to add warnings if the executables are not available ? 
Or in the diagnostic page, if optimized is set to 1 but the softwares are not there?

File: config/image-optimizer.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /*
+     * The maximum time in seconds each optimizer is allowed to run separately.
+     */
+    'timeout' => 10,
+
+    /*
+     * If set to `true` all output of the optimizer binaries will be appended to the default log.
+     * You can also set this to a class that implements `Psr\Log\LoggerInterface`.
+     */
+    'log_optimizer_activity' => false,
```
Comment (tmp-hallenser) [inline config/image-optimizer.php]: @ildyria Can we plug the Lychee Logging into here?

**Comment by tmp-hallenser** [comment]
> LGTM.
> 
> Do we want to add warnings if the executables are not available ?
> Or in the diagnostic page, if optimized is set to 1 but the softwares are not there?

Good idea. The library already provides some logging, we need to specify a logging interface (type `Psr\Log\LoggerInterface`) -> can we use the Lychee Logging?

File: config/image-optimizer.php (hunk 1 above)
Comment (ildyria) [reply config/image-optimizer.php]: I don't think we have something which implements Psr\Log\LoggerInterface but that can probably be easilly done. :)

**Comment by ildyria** [comment]
What I am also worried, is, if you enable the optimizer but the software are not here, does it raises exceptions or does it just append to the logs ?

**Comment by ildyria** [comment]
@tmp-hallenser can you double check ?

**Comment by tmp-hallenser** [comment]
works, thanks. Nice job with the logger.

**Pull request merged**